### ISSUE TITLE
Possibly awkward controls?

### ISSUE THREAD

**Issue by ELLIOTTCABLE** [lambdalisue/jupyter-vim-binding#64]
So, after #27 didn't address #47 for me, and I've had slightly more opportunity to _use_ this plugin, I'd like to mention some concerns:
1. As has been said elsewhere, `<Shift-Esc>` really _is_ a bit much, I think. It sounds like you've thought this through and are convinced, so I'll just leave this as my :-1: to boost the other voices I've heard
2. `F1` to open help doesn't seem to work on Safari / OS X …
3. `J`/`K` are really unintuitive in command-mode: `⌃J` and `⌃K` are really uncompfortable to reach for (especially on a laptop keyboard, with the `Fn`-keys pushing `Ctrl` even _further_ over, under my palm …); what is the rationale for `J`/`K` in command-mode simply scrolling? We all have built-in browser behaviours for that sort of thing … (touchpad-swiping, space / ⇧-space, etceteras.) I really think these would be better repurposed as previous/next cell.

Other than that, I think #27's been a big step forward! (=

**Comment by lambdalisue** [comment]
> As has been said elsewhere, <Shift-Esc> really is a bit much, I think. It sounds like you've thought this through and are convinced, so I'll just leave this as my :-1: to boost the other voices I've heard

Hum... Do you have any alternative idea?

> F1 to open help doesn't seem to work on Safari / OS X …

That's would be a bug. Please make a new issue for this and I'll check.

> J/K are really unintuitive in command-mode: ⌃J and ⌃K are really uncompfortable to reach for (especially on a laptop keyboard, with the Fn-keys pushing Ctrl even further over, under my palm …); what is the rationale for J/K in command-mode simply scrolling? We all have built-in browser behaviours for that sort of thing … (touchpad-swiping, space / ⇧-space, etceteras.) I really think these would be better repurposed as previous/next cell.

This is just my feeling but `j/k` should move per lines or similar. I feel `j/k` per cells beyond my expectation so I map `^-j/^-k` for per cells.

You said `^-X` mapping is uncomfortable but then how do you be comfortable when you use a native Vim? A native Vim also use a lot of `^-X` mappings so same problem should exists on a native Vim I guess.
I am satisfied with `^-J` or `^-K` while my _caps lock_ works as _ctrl_. Yes, this is my _special_ configure but I think jupyter-vim-binding should follow Vim's mapping flavor. I also think that most of Vimmer use _caps lock_ -> _ctrl_ kind of keyboard remapping.

**Comment by ELLIOTTCABLE** [comment]
Strange, I use … _almost no_ control-key vim-bindings. Off the top of my head, only ⌃Q and G-⌃G … both of which are very-special-case, and I definitely don't use on a regular basis.

(Personal, my caps-lock is remapped to Esc)

Maybe we could get somebody else to weigh in? Sounds like one of the two of us may be outliers. /=

**Comment by lambdalisue** [comment]
ctrl-w x for window operation
ctrl-n/p for completion
ctrl-o/i for jump
ctrl-y/e/d/u for scroll
ctrl-l for refresh
ctrl-h for backspace
ctrl-a/x for increment/decrement
ctrl-v for visual block
...

I'm sure Vim use ctrl in everywhere.

**Comment by ELLIOTTCABLE** [comment]
I mean, again: how many users actually use any of those? I remapped
<Leader>-W (which for me is spacebar-W) for window controls, for exactly
the same reason; almost everyone uses tab for completion, native scrollbars
or zz / zt / etc for scrolling; who actually _types_ ctrl-H instead of
sending it with their backspace key … so on, and so forth. Literally the
only use _I_ get out of the control key (save ctrl-W, which I personally
remapped, but that's a valid observation for most users probably) is ⌃V
(for me, and other physical-paste users, ⌃Q).

I think I've made my point; I'll leave it to other users to weigh-in, here?
(=

**Comment by lambdalisue** [comment]
We can not discuss how many people use ctrl mapping while everybody customize them Vim (additionally, who I know, including me, prefer default mappings so the my answer to your question is "a lot of people")
The point is these are default mapping.

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pi&cv=6.0.64&pv=8.2]
On 火, 3月 29, 2016 at 4:45 午後, ELLIOTTCABLE notifications@github.com wrote:
I mean, again: how many users actually use any of those? I remapped
<Leader>-W (which for me is spacebar-W) for window controls, for exactly
the same reason; almost everyone uses tab for completion, native scrollbars
or zz / zt / etc for scrolling; who actually _types_ ctrl-H instead of
sending it with their backspace key … so on, and so forth. Literally the
only use _I_ get out of the control key (save ctrl-W, which I personally
remapped, but that's a valid observation for most users probably) is ⌃V
(for me, and other physical-paste users, ⌃Q).

I think I've made my point; I'll leave it to other users to weigh-in, here?
(=

**Comment by ElDeveloper** [comment]
> > I'm sure Vim use ctrl in everywhere.
> 
> I mean, again: how many users actually use any of those?

I surely use most of those.

**Comment by Gamrix** [comment]
<F1> brings up the Chrome help when using Chrome on Windows.
